### ISSUE TITLE
Feature Request: allow to split health statistics by Mobile OS, Desktop OS, Browser etc

### ISSUE THREAD

**Issue by ghost** [getsentry/sentry#58348]
[copied from discussion below, since there originally wasn't any detail here]

we are a game studio company that uses unity to build our games, with unity we create our iOS version and Android version of our game. both under the same Sentry Project. in sentry, under the Releases dashboard, Discovery and Widget, we can see our health statistics for the project (crash_rate, crash_free_rate for users or for sessions)
the problem is that those statistics do not come with the information of which os.name it belongs too and there is no way to filter so we can truly understand if there is difference between Android Health and iOS health.

Request: A way to filter the release health data by OS.

┆Issue is synchronized with this [Jira Improvement](https://getsentry.atlassian.net/browse/FEEDBACK-2025) by [Unito](https://www.unito.io)

**Comment by Kobby-Bawuah** [comment]
Hello @matanplay, could you further elaborate for whoever has to work on this?

**Comment by matanplay** [comment]
@Kobby-Bawuah yes. we are a game studio company that uses unity to build our games, with unity we create our iOS version and Android version of our game. both under the same Sentry Project. in sentry, under the Releases dashboard, Discovery and Widget, we can see our health statistics for the project (crash_rate, crash_free_rate for users or for sessions)
the problem is that those statistics do not come with the information of which os.name it belongs too and there is no way to filter so we can truly understand if there is difference between Android Health and iOS health.

**Comment by hubertdeng123** [comment]
Thanks for writing in, I'll make sure the appropriate team sees this

**Comment by roggenkemper** [comment]
@matanplay just to clarify - your request is a way on the Releases page to filter by os.name in your case, so you can see the release health per os, not just per project since they both come from the same Unity project

**Comment by matanplay** [comment]
exactly.

**Comment by lobsterkatie** [comment]
@matanplay - I'm going to put this into our backlog and bring it to our PMs for consideration.

**Comment by Angelodaniel** [comment]
In addition to that I would like to be able to see the crash_free_rate on a per page basis.

**Comment by matanplay** [comment]
@Dhrumil-Sentry while we are on this request - I would like to add a bit more "filtering" to release health:
1 - filter by country/geo (ability to answer what is my crash free rate in the US alone, or India alone for a specific release and specific mobile OS)
2 - filter by user id (ability to answer what is the crash free rate for the latest sessions of a specific user)
3 - is it possible to expose DAU? as a property we can use ?

**Comment by lobsterkatie** [comment]
@matanplay - those are interesting ideas. They're not possible currently, in terms of the data we send along with crashed/non-crashed sessions, but might be possible with the metrics offering we're developing. I'll let @matejminar and @ale-cota weigh in on that score.

**Comment by Angelodaniel** [comment]
Hey @lobsterkatie @Dhrumil-Sentry, would this proposed solution on Metric still be something for the future? 
Please revisit this topic as more customers are requesting the above feature 🙏

**Comment by Dhrumil-Sentry** [comment]
There are no concrete plans to support this feature as of now. We will keep the issue to open to gauge interest from users

**Comment by mortargrind** [comment]
@Dhrumil-Sentry 
Hello! 👋 I was redirected here by our Sentry support for our org.

I can't seem to use the regular built-in tags like browser/os etc. in crash rate related widgets in Dashboards. This is a critical feature for our org.

Because we have many web apps that are rendered inside native host apps, which can be an iOS or an Android app. Since the underlying browser engine implementation and the surrounding host application implementations are different, the crash rate will be different too. 
We need crash frate for each different OS to calculate separate golden metrics for host apps. These golden metrics are quite critical for our long term strategy as the whole org. 

One hack I can think of is having 2 different Sentry projects for the same physical app to be able to use the "project" filter. But that would have an immense side effect on every other thing. The other hack might be using different release tags just for Sentry config (instead of `v1.0.0` `v1.0.0-ios`/`v1.0.0-android`), which might have a lower impact on the developer workflow.

My general feedback related to this is: Dashboarding & querying things in general should be more flexible and elastic. Every app is different and as a result any developer working on any app might need to search for a slightly different answer based on their requirements and situation. If we continue on with this instance; the inability to have OS information for release or crash related queries seems completely arbitrary from the user's perspective (I am sure there are technical reasons for this situation based on the decisions you have made on your end in the past that cannot be changed easily). It's not predictable or scalable which questions (query) we ask (execute) from our perspective.

**Comment by Dhrumil-Sentry** [comment]
Thanks a lot for this feedback. There are indeed technical reasons that make this infeasible to do today and it's not an easy lift in terms of technical effort and cost to make these tags available.

We will keep this feedback under consideration

**Comment by iamcaner** [comment]
this would be helpful for us as well to differentiate Android and iOS crash rates for a shared Flutter project.

**Comment by Angelodaniel** [comment]
Hey Team, customer I visited also explained that it's really hard to distinguish or get numbers based on platform for their game. 
Adding more context to the release data would be of great value.

**Comment by InterstellarStella** [comment]
https://sentry.zendesk.com/agent/tickets/159852

**Comment by bcoe** [comment]
Forwarding internally to the folks who have been working most recently on releases.

**Comment by InterstellarStella** [comment]
https://sentry.zendesk.com/agent/tickets/161454

**Comment by Angelodaniel** [comment]
Hey Team, a customer has brought the need of this up in the meeting today.
Basically with Unity it's great if we can split up health statistics (read Crash free rates) based on Platform. 
Having the data still inside of 1 release is a must as it's easy to loose the complete overview if you're going to split release names based on platform. So having this as a extra parameter on the "single" release itself is greatly helping them to debug their apps in a better way and even lean on Sentry more.

**Comment by bcoe** [comment]
We are currently limited by the capabilities of the release dataset.

**Comment by rodolfoBee** [comment]
Also requested in: https://sentry.zendesk.com/agent/tickets/162274

**Comment by minori-fh** [comment]
relevant internal ticket: https://sentry.zendesk.com/agent/tickets/167458

**Comment by iftheshoefritz** [comment]
> We are currently limited by the capabilities of the release dataset

@bcoe could you elaborate on "limitations of the release dataset"? Is that limitation something that can only be improved by internal changes? Can users submit more data when creating releases? (I've searched for the latter in your docs and I can't find anything)

**Comment by bcoe** [comment]
> @bcoe could you elaborate on "limitations of the release dataset"?

Our release dataset is not using our modern database (the one used for spans, metrics, logs, and _soon_ errors) that allows for querying and grouping on arbitrary attributes. So, today, we can't let you add an arbitrary attribute to a release, e.g., `user.geo.country_code`, and then let you query only release sessions with that attribute set.

There is an interest in modernizing this dataset to support this, but it requires backend work not scheduled yet.
